src/components/Search/Filter.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react'
+import {
+  Text,
+  StyleSheet,
+  ScrollView,
+  Pressable,
+  PressableProps,
+  ScrollViewProps,
+} from 'react-native'
+import Icon from 'react-native-vector-icons/Entypo'
+
+import Colours from '../../themes/colours'
+import useToggle from '../../hooks/use-toggle'
+import { px, py } from '../../themes/utils'
+
+type FilterStripProps = {
+  /**
+   * Search filters user can choose from.
+   */
+  availableFilters: string[],
+} & Partial<
+Omit<ScrollViewProps, 'showsHorizontalScrollIndicator' | 'horizontal'>
+>
+
+type FilterButtonProps = {
+  /**
+   * Value Filter Button should display.
+   */
+  label: string,
+} & Omit<PressableProps, 'children'>
+
+const styles = StyleSheet.create( {
+  crossIcon: {
+    paddingLeft: 5,
+    marginRight: -5,
+  },
+  option: {
+    ...py( 2 ),
+    ...px( 10 ),
+    marginRight: 5,
+    borderRadius: 15,
+    display: 'flex',
+    flexDirection: 'row',
+  },
+  optionText: {
+    fontSize: 10,
+    fontWeight: '500',
+  },
+} )
+
+export const FilterButton = ( { label, ...props }: FilterButtonProps ) => {
+  const { state, toggle } = useToggle( false )
+
+  const optionDynamicStyles = StyleSheet.flatten( [
```

Comment:
I would keep styles out of here and change `styles` to `getStyles`, supplying any parameters you wish to use in an object. I think we need a solution which borrows from `css-in-js`'s `useStyles` hook, but for now, the aforementioned will work.

Also, reading https://github.com/DefinitelyTyped/DefinitelyTyped/issues/29265#issuecomment-430783289 seems to indicate there's no real reason to use `StyleSheet.create` anymore 🤷‍♀️

Reply:
I don't think I understand it properly how to do this
> I would keep styles out of here and change styles to getStyles, supplying any parameters you wish to use in an object. 

Do you mean something like this and then I pass it since what I understand from that issue this is valid and there are no difference?
```ts
const optionDynamicStyles = {
  ...styles.option,
  backgroundColor: state ? Colours.MediumLightGray : Colours.DarkGray,
}
```